src/main/java/com/google/sps/image/ImageSelection.java

```diff
@@ -90,18 +98,53 @@ private static float findBestImage(ImageScorer imageScorer, StringBuilder bestIm
     return bestImageScore;
   }
 
+  private List<String> addBackups(List<String> images, int noBackups) {
+    for (int i = 1; i <= noBackups; ++i) {
+      switch(i) {
```

Comment:
well, this seems an overkill, you can just use BACKUP_IMAGE of i inside the for
String.format("BACKUP_IMAGE%d", i);
number of backups needed is anyway fixed based on EXTRACTED_IMAGES

Reply:
Thanks for this nice solution!

Reply:
Java requires to know in advance variable names, thus is not really possible to compute variable name in runtime, it's possible however using reflection or a map or an array.
Reflection however is usable only if your object is a class, which is not your case, and it doesn't make sense to create one for this use case.
You can simplify this with a map or an array

Reply:
Done by using an array.